src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -528,14 +528,22 @@ private static bool IsIntPtrToNativeIntegerNestedCast(ExpressionSyntax castNode,
 
                 // Given (nuint)(nint)myIntPtr we would normally suggest removing the (nint) cast as being identity
                 // but it is required as a means to get from IntPtr to nuint, and vice versa from UIntPtr to nint,
-                // so we check for an identity cast from [U]IntPtr to n[u]int, and a parent cast to the opposite.
+                // so we check for an identity cast from [U]IntPtr to n[u]int
                 if (castedExpressionType.SpecialType == castType.SpecialType &&
                     !castedExpressionType.IsNativeIntegerType &&
-                    castType.IsNativeIntegerType &&
-                    parentCastType.IsNativeIntegerType &&
-                    parentCastType.SpecialType == oppositeType)
+                    castType.IsNativeIntegerType)
                 {
-                    return true;
+                    // Do we have a parent cast to the opposite?
+                    if (parentCastType.IsNativeIntegerType &&
+                        parentCastType.SpecialType == oppositeType)
+                    {
+                        return true;
+                    }
+                    // cases like (int)(nint)myIntPtr
+                    if (parentCastType.IsNumericType())
```

Comment:
This is probably too broad. `IsNumeric` applies to
```CS
                switch (type.SpecialType)
                {
                    case SpecialType.System_Byte:
                    case SpecialType.System_SByte:
                    case SpecialType.System_Int16:
                    case SpecialType.System_UInt16:
                    case SpecialType.System_Int32:
                    case SpecialType.System_UInt32:
                    case SpecialType.System_Int64:
                    case SpecialType.System_UInt64:
                    case SpecialType.System_Single:
                    case SpecialType.System_Double:
                    case SpecialType.System_Decimal:
                    case SpecialType.System_IntPtr when type.IsNativeIntegerType:
                    case SpecialType.System_UIntPtr when type.IsNativeIntegerType:
                        return true;
                }
```
It also makes the previous check `parentCastType.IsNativeIntegerType && parentCastType.SpecialType == oppositeType` redundant (oppositeType is either IntPtr or UIntPtr).

Reply:
Being too broad is ok here. Better to leave in some unnecessary casts than remove some necessary ones.